web/pdf_page_view.js

```diff
@@ -511,16 +507,6 @@ class PDFPageView {
       this.paintedViewportMap.delete(this.svg);
       delete this.svg;
     }
-
-    if (!this.loadingIconDiv) {
-      this.loadingIconDiv = document.createElement("div");
-      this.loadingIconDiv.className = "loadingIcon notVisible";
-      this.loadingIconDiv.setAttribute("role", "img");
-      this.l10n.get("loading").then(msg => {
-        this.loadingIconDiv?.setAttribute("aria-label", msg);
-      });
-      div.append(this.loadingIconDiv);
-    }
```

Comment:
Quick local testing seem to indicate that https://github.com/mozilla/pdf.js/pull/15992#issuecomment-1413583272 can be fixed by placing the `div.classList.add("loadingIcon");` line here instead of below.

Reply:
Why is that a problem, since it'd *considerably* simplify the implementation?

Reply:
Because the pseudo elements are traversed on restyle.
You can verify that in using the Firefox profiler and in looking at the tab `Marker Chart` at the line `Styles` (I tested with `pdf.pdf`).
Not having the pseudo elements divides by 2 the time spent to restyle when zooming. So from a performance point of view it's better like this.

Anyway if you've some ideas to keep the perf improvement and at the same time to simplify the implementation, I take.